File: LC45.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        far = 0
        end = 0
        cnt = 0
        n = len(nums)

        for i in range(n-1):
            far = max(far, i + nums[i])

            if i == end:
                cnt += 1
                end = far
        
        return cnt

File: LC45.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int cnt = 0, far = 0, end = 0, n = nums.size();

        for(int i = 0; i < n-1; i++){
            far = max(far, i + nums[i]);

            if(i == end){
                cnt++;
                end = far;
            }
        }

        return cnt;
    }
};
